### PR TITLE
integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Harrynoqs**
Write integration tests with Capybara gem.
Secure app from n+1 problems.
217371156-d407879b-7bd8-46fc-a79a-58e270b4e59f
217371163-2d8fad90-02c5-452a-800e-9035b23b0a7b

### CODE REVIEW

File: spec/system/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
```
Comment (rloterh) [inline spec/system/posts_index_spec.rb:2]: - [ ]  A number of the spec test cases are failing, kindly ensure all test cases are passing.

**Failing test cases:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Harry, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦   🏃

## Highlights 👍  
- N + 1 problem addressed!
- Professional ReadMe!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- Kindly include appropriate instructions in your ReadMe on how to run your app, under the _Getting Started_ section.
- Kindly endeavor to remove the guide note below the MIT license.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/system/posts_index_spec.rb (hunk 1 above)
Comment (Harrynoqs) [reply spec/system/posts_index_spec.rb:2]: done

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**